packages/expo-splash-screen/android/src/main/java/expo/modules/splashscreen/SplashScreenModule.kt

```diff
@@ -29,6 +33,24 @@ class SplashScreenModule(context: Context) : ExportedModule(context) {
 
   override fun onCreate(moduleRegistry: ModuleRegistry) {
```

Comment:
Is is invoked in the very same moment that `Activity#onCreate` is? 🤔 
I mean, we need to be sure that there is no flashing in-between platform native splash screen handled by the Android framework & our custom view that is mounted on the activity.

Reply:
It's called sometime after `Activity#onCreate` not sure how much longer given there isn't much documentation around how unimodules execute this delegated method. I'd imagine this is equivalent to iOS AppDelegate proxy.

Reply:
@tsapeta - do you know the answer to @bbarthec's question above?

Reply:
It's being called during the module initialization, that is when the module registry creates and registers an instance of the module. It has nothing to do with `Activity#onCreate`. It's actually the same as `setModuleRegistry:` on iOS.

Reply:
According to what @tsapeta wrote it would cause quite a problem 😞
What we want and what we have: `SplashScreenView` has to be mounted and presented immediately after `setContentView` is called on the Activity. This ensures us that there are no screens we do not have control of.
What this PR is doing: `SplashScreenView` is mounted as soon as the module registry is initialized and this is a while later 😞 This leaves us with the scenario: native Android splash view (hides immediately after `setContentView`, which is invoked somewhere in `Activity.onCreate`) ➡️ `ReactRootView` container with no content ➡️ `SplashScreenView` is mounted once the `moduleRegistry` catches up.

We are lacking some mechanism that hooks into activities lifecycle and propagates the signals to our modules 🤔
Maybe there are some new ways to achieve this, given that Jetpack/Kotlin comes with variety of lifecycle-aware libs.